### PR TITLE
GCW- 2666: Provide filters for offer search UI

### PR DESCRIPTION
**Pull request by abulsayyad123**
Hi Team,
This PR is for ticket GCW-26666- Filter UI for searching offers. I have added 3 filters button for State, Reviewer and Time Due along with test cases.
Please review the PR and let me know if any change is required.

### CODE REVIEW

File: app/helpers/state-filter-icon.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  REVIEWING,
+  REVIEWED,
+  SCHEDULED,
+  RECEIVING,
+  RECEIVED,
+  CANCELLED,
+  INACTIVE
+} = STATE_FILTERS;
+
+export default Ember.Helper.helper(function(state) {
```
Comment (patrixr) [inline app/helpers/state-filter-icon.js]: maybe a mapping would be simpler 

```javascript
const STATE_ICONS = {
   [NEW]: "shopping-basket",
   [REVIEWING]: "list-ol",
   ...
};

export default Ember.Helper.helper(function(state) {
   return STATE_ICONS[state] || "";
});
```

File: app/helpers/state-filter-icon.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import Ember from "ember";
+import { STATE_FILTERS } from "../services/filter-service";
+
+const {
```
Comment (patrixr) [inline app/helpers/state-filter-icon.js]: nice, referencing filters from other file 👍

File: app/locales/zh-tw/translations.coffee [hunk 3]
```diff
@@ -65,6 +65,39 @@ I18nTranslationsZhTw =
...
+        week: "This week",
+        next_week: "Next week",
+        month: "This month",
+        next_month: "Next month"
+
+      self_review: "Mine"
+      all_review: "All"
+      time_filter_title: "Offer Due Date"
+      button_state: "State"
+      showPriority: "Show Priority"
```
Comment (patrixr) [inline app/locales/zh-tw/translations.coffee]: I think some translations are already in Stock, maybe you can copy them over 

I notice this in stock
```
    showPriority: "只顯示優先處理訂單",
    showPriority_info: "優先訂單已停留在一狀態過長，需要盡快處理",
```

File: app/controllers/search.js [hunk 4]
```diff
@@ -34,59 +37,103 @@ export default Ember.Controller.extend(backNavigator, {
...
-        this.set("filteredResults", results);
-      })
-      .finally(() => {
-        loadingView.destroy();
-      });
+  getFilterQuery() {
+    const filterService = this.get("filterService");
+    const isPriority = filterService.isPriority();
+    const { after, before } = filterService.get("offerTimeRange");
+    let stateFilters = _.without(
```
Comment (patrixr) [inline app/controllers/search.js]: I don't think `"showPriority"` should be mixed with the state filters.
It doesn't make sense and forces those kinds of workarounds. `showPriority` should be a boolean we toggle.

This is copied from stock, no need to change it now.
But if anyone has an opportunity/bandwidth one day it'd be nice to clear it up :) 

cc - @abhinav098 @namrataukirde @abulsayyad123

File: app/components/offer-filters-button.js [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    redirectTofilters(queryParam) {
+      const offerFilter = {};
+      offerFilter[queryParam] = true;
+      this.get("router").transitionTo("offers_filters", {
+        queryParams: offerFilter
+      });
+    },
+
+    clearStateFilters() {
+      this.get("filterService").clearOfferStateFilters();
```
Comment (patrixr) [inline app/components/offer-filters-button.js]: I recently discovered that we can pass a `target` to the actions in `.hbs` files
Meaning we can do `{{ action 'enableReviewFilter' target=filterService}}` , which means we don't have to create those kinds of proxy actions.

Not a problem having them, just a good to know : )

File: app/templates/_state_filters.hbs [hunk 6]
```diff
@@ -0,0 +1,28 @@
+<div class="state-filter-block">
+  {{!-- Offer Priority Checkbox --}}
+  <div class="row">
+    <div class="state-filter-title small-12 medium-12 large-12 columns">
+      Only Show Priority Offers
```
Comment (abhinav098) [inline app/templates/_state_filters.hbs]: please add this to translations.

File: app/templates/_state_filters.hbs [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+    <div class='checkbox-area small-12 medium-12 large-12 columns'>
+      {{offer-filter-checkbox name="showPriority"}}
+    </div>
+  </div>
+  <br/>
+
+  {{!-- Offer Status Block --}}
+  <div class="row">
+    <div class="state-filter-title small-12 medium-12 large-12 columns">
+      Offer Statuses
```
Comment (abhinav098) [inline app/templates/_state_filters.hbs]: same as above

File: app/templates/_time_filters.hbs [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+  <div class="row">
+    <div class="time-filter-title small-12 medium-12 large-12 columns">
+      {{t "offer_filters.time_filter_title"}}
+    </div>
+  </div>
+
+  <div class="row time_inputs">
+      <div class=" small-6 columns date_selector">
+        <div class="small-12 columns type-of-id-text">
+          On or after
```
Comment (abhinav098) [inline app/templates/_time_filters.hbs]: translations please. :)

File: app/templates/_time_filters.hbs [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+            name='selectedTimeRangeAfter'
+            id='selectedTimeRangeAfter'
+            model=selectedTimeRange.after
+            onSelect=(action 'setAfterTime')
+          }}
+          <i class="fa fa-calendar calender_icon"></i>
+        </div>
+      </div>
+      <div class=" small-6 columns date_selector">
+        <div> On or before </div>
```
Comment (abhinav098) [inline app/templates/_time_filters.hbs]: same as above

File: app/templates/components/offer-filters.hbs [hunk 10]
```diff
@@ -0,0 +1,16 @@
+<div class="offer-filters">
+  <div class="row">
+    <div class="small-6 columns">
+      <button {{action "applyFilters"}} class="button expand filter-btn apply">Apply</button>
```
Comment (abhinav098) [inline app/templates/components/offer-filters.hbs]: translations here too.

File: app/templates/components/offer-filters.hbs [hunk 11]
```diff
@@ -0,0 +1,16 @@
...
+    </div>
+    <div class="small-6 columns">
+      <button {{action "clearFilters"}} class="button expand filter-btn clear">Clear</button>
```
Comment (abhinav098) [inline app/templates/components/offer-filters.hbs]: and here.

File: app/templates/search.hbs [hunk 12]
```diff
@@ -1,31 +1,38 @@
...
-    {{focus-textfield
-      name="searchText"
-      id="searchText"
-      placeholder=(t 'search.placeholder')
-      value=searchText }}
+<div class="row search-box ui">
+  <div class="row">
+    <div class="small-2 columns back-icon" {{action "cancelSearch"}}>
+      <i class="fa fa-angle-left"></i>
+      <span>Back</span>
```
Comment (abhinav098) [inline app/templates/search.hbs]: translations for this "Back" too.

File: app/helpers/state-filter-icon.js (hunk 1 above)
Comment (abulsayyad123) [reply app/helpers/state-filter-icon.js]: This looks nice 👌

File: app/components/offer-filters-button.js (hunk 5 above)
Comment (abulsayyad123) [reply app/components/offer-filters-button.js]: This is awesome !!

File: app/templates/_state_filters.hbs (hunk 7 above)
Comment (abulsayyad123) [reply app/templates/_state_filters.hbs]: added

File: app/templates/_state_filters.hbs (hunk 6 above)
Comment (abulsayyad123) [reply app/templates/_state_filters.hbs]: added

File: app/templates/_time_filters.hbs (hunk 9 above)
Comment (abulsayyad123) [reply app/templates/_time_filters.hbs]: added

File: app/templates/components/offer-filters.hbs (hunk 10 above)
Comment (abulsayyad123) [reply app/templates/components/offer-filters.hbs]: added

File: app/templates/search.hbs (hunk 12 above)
Comment (abulsayyad123) [reply app/templates/search.hbs]: added

File: app/templates/components/offer-filters.hbs (hunk 11 above)
Comment (abulsayyad123) [reply app/templates/components/offer-filters.hbs]: added

**Pull request merged**